internal/consts/finalizers.go

```diff
@@ -33,7 +33,13 @@ const (
 	// DataImportJobRBACCleanupFinalizer is a finalizer that is used to clean up cluster-wide RBAC resources.
 	// This is needed because DataImportJob cannot own cluster-wide resources like ClusterRole and ClusterRoleBinding,
 	// as Kubernetes does not allow cluster-scoped resources to be owned by namespace-scoped resources.
+	//
+	// Deprecated: Use ordered-cleanup finalizer instead. This finalizer is kept for backward compatibility.
```

Comment:
I've kept this finalizer assuming we will deliver this fix in 1.9.0. If we agree to deliver it in 1.8.0, then we can remove this finalizer completely